src/main/java/seedu/address/logic/parser/FindEventCommandParser.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_FRIEND_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import seedu.address.logic.commands.FindEventCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.event.EventFilterPredicate;
+
+/**
+ * Parses input arguments and creates a new FindEventCommand object
+ */
+public class FindEventCommandParser implements Parser<FindEventCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindEventCommand
+     * and returns a FindEventCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindEventCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DATE, PREFIX_FRIEND_NAME);
+
+        boolean isAnyArgumentGiven = argMultimap.arePrefixesPresent(PREFIX_NAME)
+                || argMultimap.arePrefixesPresent(PREFIX_DATE)
+                || argMultimap.arePrefixesPresent(PREFIX_FRIEND_NAME);
+
+        if (!isAnyArgumentGiven || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindEventCommand.MESSAGE_USAGE));
+        }
+
+        String eventNameSubstring;
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            eventNameSubstring = argMultimap.getValue(PREFIX_NAME).get();
+        } else {
+            eventNameSubstring = "";
```

Comment:
Why not let this be `null`? Then you can handle default values in the constructor of `EventFilterPredicate`